src/Components/Layout.tsx

```diff
@@ -1,596 +1,634 @@
-              <Button
-                onClick={() => applyChanges()}
-                variant="contained"
-                color="primary"
-                className={classes.headerApplyButton}
-              >
-                {getApplyText()}
-              </Button>
-            </Box>
-            <Tab>
-              <Box ml={2}>
-                <Button color="primary" variant="text">
-                  {dateRange.finalDisplayText[0]}
-                </Button>
-              </Box>
-            </Tab>
-            <span>&#10230;</span>
-            <Tab>
-              <Button color="primary" variant="text">
-                {dateRange.finalDisplayText[1]}
-              </Button>
-            </Tab>
-          </TabList>
-          <TabPanel>
-            <QuickSelect {...getQuickSelectObj()} />
-          </TabPanel>
-          <TabPanel>
-            <MenuView {...getMenuObj()} />
-          </TabPanel>
-          <TabPanel>
-            <Body {...getBodyObj(0)} {...getDateSelectObj()} />
-          </TabPanel>
-          <TabPanel>
-            <Body {...getBodyObj(1)} {...getDateSelectObj()} />
-          </TabPanel>
-        </Tabs>
-      </div>
-    </MuiThemeProvider>
-  );
-};
+import React, { useState } from "react";
+import { Button, Box } from "@material-ui/core";
+import ms from "ms";
+
+import { DateRange } from "./DateRange.tsx";
+import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import { Body } from "./Body.tsx";
+import { MenuView } from "./Menu.tsx";
+import { TimerUI } from "./Timer.tsx";
+import { QuickSelect } from "./QuickSelect.tsx";
+import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
+import KeyboardTabIcon from "@material-ui/icons/KeyboardTab";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import RefreshIcon from "@material-ui/icons/Refresh";
+import { makeStyles } from "@material-ui/styles";
+import "./Styling.css";
+
+var locale = require("browser-locale")();
+const humanize = require("humanize-duration");
+
+interface Inputs {
+  resetFn(): void;
+  getDateRange(dates: Date[]): void;
+  dateFormatter?: Intl.DateTimeFormat;
+  theme?: any;
+  commonlyUsedText?: number[];
+  quickSelectTerms?: string[];
+  quickSelectIntervals?: number[];
+  relativeTerms?: string[];
+  relativeIntervals?: number[];
+  timeFormat?: string;
+  nowText?: string;
+  minimumYearValue?: number;
+  maximumYearValue?: number;
+  humanizer?: any;
+  setStoredRange?(dateRange: string[]): void;
+  storedRange: string[] | null;
+}
+
+const commonlyUsedTextDefault: number[] = [
+  -ms("15 Minutes"),
+  -ms("30 Minutes"),
+  -ms("1 Hour"),
+  -ms("24 hours"),
+  -ms("7 days"),
+  -ms("30 days"),
+  -ms("90 days"),
+  -ms("1 year"),
+];
+
+const quickSelectTermsDefault: string[] = ["Last", "Next"];
+
+const timeFormatDefault: string = locale.slice(0, 2);
+
+const quickSelectIntervalsDefault: number[] = [
```

Comment:
Pull all of these defaults and constants out into a Constants.ts file so that your main Components.ts file gets cleaner and easier to read.